File: MasterDirectory/MasterDirectory.Web/Modules/Consultorios/CatalogosConsultorios/CatalogosConsultoriosForm.cs
using Serenity.ComponentModel;
using System;
using System.ComponentModel.DataAnnotations;

namespace MasterDirectory.Consultorios.Forms;

[FormScript("Consultorios.CatalogosConsultorios")]
[BasedOnRow(typeof(CatalogosConsultoriosRow), CheckNames = true)]
public class CatalogosConsultoriosForm
{
    [Editable(false)]
    public int IdtipoCatalogo { get; set; }
    [Editable(false)]
    public int IdClave { get; set; }
    public string Descripcion { get; set; }
    [BooleanEditor]
    public int Activo { get; set; }
    //public DateTime DtRegistro { get; set; }
}

File: MasterDirectory/MasterDirectory.Web/Modules/ServerTypes/Consultorios/CatalogosConsultoriosForm.ts
import { LookupEditor, IntegerEditor, StringEditor, BooleanEditor, PrefixedContext } from "@serenity-is/corelib";
import { initFormType } from "@serenity-is/corelib/q";

export interface CatalogosConsultoriosForm {
    IdtipoCatalogo: LookupEditor;
    IdClave: IntegerEditor;
    Descripcion: StringEditor;
    Activo: BooleanEditor;
}

export class CatalogosConsultoriosForm extends PrefixedContext {
    static formKey = 'Consultorios.CatalogosConsultorios';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!CatalogosConsultoriosForm.init)  {
            CatalogosConsultoriosForm.init = true;

            var w0 = LookupEditor;
            var w1 = IntegerEditor;
            var w2 = StringEditor;
            var w3 = BooleanEditor;

            initFormType(CatalogosConsultoriosForm, [
                'IdtipoCatalogo', w0,
                'IdClave', w1,
                'Descripcion', w2,
                'Activo', w3
            ]);
        }
    }
}